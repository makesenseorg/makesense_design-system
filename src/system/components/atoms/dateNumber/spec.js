/* eslint-disable no-undef */
import { shallowMount } from "@vue/test-utils";
import Comp from "./DateNumber.vue";

describe("DateNumber.vue", () => {
  it("matches snapshot", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        number: 1234,
      },
    });
    expect(wrapper.element).toMatchSnapshot();
  });

  it("has primary as default color class", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        number: 1234,
      },
    });
    expect(wrapper.classes()).toContain("date--color-primary");
  });

  it("sets class with color modifier", () => {
    const wrapper = shallowMount(Comp, {
      propsData: {
        number: 1234,
        color: "secondary",
      },
    });
    expect(wrapper.classes()).toContain("date--color-secondary");
  });

  it("matches snapshot with slot", () => {
    const wrapper = shallowMount(Comp, {
      slots: {
        default: "<div>2000</div>",
      },
      propsData: {},
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
