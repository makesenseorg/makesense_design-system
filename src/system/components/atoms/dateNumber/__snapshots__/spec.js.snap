// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateNumber.vue matches snapshot 1`] = `
<div
  class="date date--color-primary"
>
  <div
    class="date__number"
  >
    
      1234
    
  </div>
</div>
`;

exports[`DateNumber.vue matches snapshot with slot 1`] = `
<div
  class="date date--color-primary"
>
  <div
    class="date__number"
  >
    <div>
      2000
    </div>
  </div>
</div>
`;
